app/code/Magento/CatalogRule/Model/Indexer/IndexerTableSwapper.php

```diff
@@ -49,7 +49,7 @@ private function createTemporaryTable(string $originalTableName): string
 
         $this->resourceConnection->getConnection()->query(
             sprintf(
-                'create table %s like %s',
+                'create temporary table %s like %s',
                 $temporaryTableName,
```

Comment:
Can I ask you slightly improve this 
```suggestion
       $this->resourceConnection->getConnection()->createTemporaryTableLike(
            $temporaryTableName,
            $this->resourceConnection->getTableName($originalTableName)
        );
```